### PR TITLE
New package: eww-0.6.0

### PR DESCRIPTION
**Pull request by shmishtopher**
#### Testing the changes
- I tested the changes in this PR: **briefly**

#### New package
- This new package conforms to the [package requirements](https://github.com/void-linux/void-packages/blob/master/CONTRIBUTING.md#package-requirements): **YES**

#### Local build testing
- I built this PR locally for my native architecture, x86_64-glibc
- I built this PR locally for these architectures:
  - aarch64-musl (crossbuild)
  - armv7l (crossbuild)
  - armv6l-musl (crossbuild)

### CODE REVIEW

**Comment by shmishtopher** [comment]
As of version `0.5.0`, eww no longer depends on nightly rust, a blocker for #37234

**Comment by classabbyamp** [comment]
commit message should be `New package: eww-0.5.0`

**Comment by shmishtopher** [comment]
My apologies, should be corrected now

File: srcpkgs/eww/template [hunk 1]
```diff
@@ -0,0 +1,18 @@
...
+makedepends="gtk+3-devel gtk-layer-shell-devel pango-devel gdk-pixbuf-devel cairo-devel glib-devel"
+short_desc="ElKowar's Wacky Widgets - a toolkit for creating custom bars and widgets"
+maintainer="Christopher K. 'Shmish' Schmitt <me@shmish.dev>"
+license="MIT"
+homepage="https://github.com/elkowar/eww"
+distfiles="https://github.com/elkowar/eww/archive/refs/tags/v${version}.tar.gz"
+checksum=ea4f62e48e3750a361e0f359933d7d840d158592ff5b3683ba1f3ccf42bda819
+
+post_install() {
+	vlicense LICENSE
```
Comment (classabbyamp) [inline srcpkgs/eww/template:17]:
```suggestion
	vlicense LICENSE
	for sh in bash fish zsh; do
		vtargetrun "${DESTDIR}/usr/bin/eww" shell-completions -s "$sh" > "eww.$sh"
		vcompletion "eww.$sh" "$sh"
	done
```

File: srcpkgs/eww/template [hunk 2]
```diff
@@ -0,0 +1,18 @@
+# Template file for 'eww'
+pkgname=eww
+version=0.5.0
+revision=1
+build_style=cargo
```
Comment (classabbyamp) [inline srcpkgs/eww/template:5]:
```suggestion
build_style=cargo
build_helper="qemu"
```
allows us to use `vtargetrun` below

File: srcpkgs/eww/template [hunk 3]
```diff
@@ -0,0 +1,18 @@
...
+make_install_args="--path=crates/eww"
+hostmakedepends="pkg-config glibc-devel libgcc-devel"
+makedepends="gtk+3-devel gtk-layer-shell-devel pango-devel gdk-pixbuf-devel cairo-devel glib-devel"
+short_desc="ElKowar's Wacky Widgets - a toolkit for creating custom bars and widgets"
```
Comment (classabbyamp) [inline srcpkgs/eww/template:9]:
```suggestion
short_desc="Toolkit for creating custom bars and widgets"
```

File: srcpkgs/eww/template [hunk 4]
```diff
@@ -0,0 +1,18 @@
+# Template file for 'eww'
+pkgname=eww
+version=0.5.0
+revision=1
+build_style=cargo
+make_install_args="--path=crates/eww"
+hostmakedepends="pkg-config glibc-devel libgcc-devel"
```
Comment (classabbyamp) [inline srcpkgs/eww/template:7]:
```suggestion
hostmakedepends="pkg-config"
```
do not appear to be needed

File: srcpkgs/eww/template [hunk 5]
```diff
@@ -0,0 +1,18 @@
...
+makedepends="gtk+3-devel gtk-layer-shell-devel pango-devel gdk-pixbuf-devel cairo-devel glib-devel"
```
Comment (classabbyamp) [inline srcpkgs/eww/template:8]: wrap at ~80 chars
```suggestion
makedepends="gtk+3-devel gtk-layer-shell-devel pango-devel gdk-pixbuf-devel 
 cairo-devel glib-devel"
```

File: srcpkgs/eww/template (hunk 1 above)
Comment (classabbyamp) [reply srcpkgs/eww/template:17]: this also requires [this patch](https://github.com/elkowar/eww/pull/1029)

```
mkdir srcpkgs/eww/patches
curl -L https://github.com/elkowar/eww/pull/1029.patch > srcpkgs/eww/patches/fix-completions.patch
```

File: srcpkgs/eww/template [hunk 6]
```diff
@@ -0,0 +1,18 @@
...
+short_desc="ElKowar's Wacky Widgets - a toolkit for creating custom bars and widgets"
+maintainer="Christopher K. 'Shmish' Schmitt <me@shmish.dev>"
+license="MIT"
+homepage="https://github.com/elkowar/eww"
```
Comment (classabbyamp) [inline srcpkgs/eww/template:12]:
```suggestion
homepage="https://elkowar.github.io/eww"
changelog="https://raw.githubusercontent.com/elkowar/eww/master/CHANGELOG.md"
```

**Comment by shmishtopher** [comment]
Requested changes have been applied

**Review by zen0bit — APPROVED** [review]
LGTM

**Comment by shmishtopher** [comment]
A new release has been made, I've updated the commit message and title to match.  `0.6.0` no longer requires a patch to generate shell completions and introduces a new `libdbusmenu-gtk3-devel` dependency.

**Comment by AnInternetTroll** [comment]
Tested on sway with the default configuration and works as expected.
Fancy looking bar

**Review by zen0bit — APPROVED** [review]

**Comment by FriederHannenheim** [comment]
Nice. Thanks for this

**Pull request merged**